apps/web/src/services/analytics/mixpanel.ts

```diff
@@ -0,0 +1,152 @@
+import mixpanel from 'mixpanel-browser'
+import type { SafeAppData } from '@safe-global/safe-gateway-typescript-sdk'
+import { IS_PRODUCTION } from '@/config/constants'
+import { DeviceType } from './types'
+import { MixPanelEventParams } from './mixpanel-events'
+import packageJson from '../../../package.json'
+
+const commonEventParams = {
+  [MixPanelEventParams.APP_VERSION]: packageJson.version,
+  [MixPanelEventParams.BLOCKCHAIN_NETWORK]: '',
+  [MixPanelEventParams.DEVICE_TYPE]: DeviceType.DESKTOP,
+  [MixPanelEventParams.SAFE_ADDRESS]: '',
+  [MixPanelEventParams.EOA_WALLET_LABEL]: '',
+  [MixPanelEventParams.EOA_WALLET_ADDRESS]: '',
+  [MixPanelEventParams.EOA_WALLET_NETWORK]: '',
+}
+
+let isMixPanelInitialized = false
+
+export const mixpanelInit = (): void => {
+  if (typeof window === 'undefined' || isMixPanelInitialized) return
+
+  const token = process.env.NEXT_PUBLIC_MIXPANEL_TOKEN
+  if (!token) {
+    if (!IS_PRODUCTION) {
+      console.warn('[MixPanel] - No token provided')
+    }
+    return
+  }
+
+  try {
+    mixpanel.init(token, {
+      debug: !IS_PRODUCTION,
+      persistence: 'localStorage',
+      autocapture: false,
+      batch_requests: true,
+      ip: false,
+    })
+
+    isMixPanelInitialized = true
+
+    if (!IS_PRODUCTION) {
+      console.info('[MixPanel] - Initialized')
+    }
+  } catch (error) {
+    console.error('[MixPanel] - Initialization failed:', error)
+  }
+}
+
+export const mixpanelSetBlockchainNetwork = (networkName: string): void => {
+  commonEventParams[MixPanelEventParams.BLOCKCHAIN_NETWORK] = networkName
+
+  if (isMixPanelInitialized) {
+    mixpanel.register({ [MixPanelEventParams.BLOCKCHAIN_NETWORK]: networkName })
+  }
```

Comment:
We are repeating the same if check in nearly every function here. How about if we abstract this into a "safeMixPanelRegister" function something like:
 

> const safeMixPanelRegister = (properties: Record<string, any>): void => {
>     if (isMixPanelInitialized) {
>       mixpanel.register(properties);
>     }
>   };

and there you do the isMixPanelInitialized check? 
Here you would just do:
safeMixPanelRegister({....})
then you won't have to do the if (...) check in every function